android/src/main/java/com/novoda/notils/devicedetection/AmazonDetector.java

```diff
@@ -0,0 +1,17 @@
+package com.novoda.notils.devicedetection;
+
+class AmazonDetector {
+
+    //From: https://developer.amazon.com/appsandservices/solutions/devices/kindle-fire/specifications/01-device-and-feature-specifications
+    private static final String AMAZON = "Amazon";
+
+    private final String buildManufacturer;
+
+    AmazonDetector(String buildManufacturer) {
```

Comment:
Consider that this class is package protected and not part of the public API.

For the second scenario one way to do it would be to pass an array of manufacturers instead of just one value ;)

Also this is in sync with the `X86Detector` that receives a bunch of values at constructor time that are calculated based on the platform. And we only want to do that once and not every time that we check.